models/res_partner.py

```diff
@@ -0,0 +1,240 @@
+        if self.error_log:
+            self.response_data.update(is_error=is_error,
+                                      error_log=self.error_log)
+            return True
+        return False
+
+    def convert_timestamp_param(self, **kwargs):
+        """Converts parameter from timestamp in string by parameter key
+         :param kwargs: dictionary with value for conversion
+         :return: str in DATETIME_FORMAT
+        """
+        timestamp = kwargs.get('value')
+        try:
+            return datetime.fromtimestamp(timestamp).strftime(DATETIME_FORMAT)
+        except Exception as error:
+            _logger.error(error)
+            self.error_log.append(str(error))
+            self.error_hendler()
+
+    @api.model
+    def civicrm_sync(self, input_params):
+        """Synchronizes ODOO and CiviCRM.
+         Creates or updates res.partner and returns response
+         :param input_params: dict of data:{
+                                'is_company': bool,
+                                'x_civicrm_id': int,
+                                'name': str,
+                                'display_name': str,
+                                'title': str,
+                                'street': str,
+                                'street2': str,
+                                'city': str,
+                                'zip': str,
+                                'country_iso_code': str,
+                                'website': str,
+                                'phone': str,
+                                'mobile': str,
+                                'fax': str,
+                                'email': str,
+                                'create_date': int,
+                                'write_date': int,
+                                'active': bool,
+                                'customer': bool
+                                }
+         :return: data in dictionary format: {
+                                'is_error': int, 0 when successful and 1 when failed
+                                'error_log': str, present when is_error is 1 and should catch the error information
+                                'contact_id': int, the id of the synced contact record
+                                'partner_id': int, the id of the corresponding partner record in Odoo
+                                'timestamp': float, the timestamp when the respond is made
+                                }
+        """
+        self.error_log = []
+        self.response_data = {'is_error': 0}
+        self.vals = input_params
+        if not self.validate_input_params(self.vals):
+            return self.get_response()
+
+        x_civicrm_id = self.vals.get('x_civicrm_id')
+
+        # Build response dictionary
+        self.response_data.update(contact_id=x_civicrm_id)
+
+        # Check if CiviCRM contact's id exists in ODOO
+        partner = self.search([('x_civicrm_id', '=', x_civicrm_id)])
```

Comment:
odoo ORM search method only search for records with active field = True, so I think you need to update the method to search for both records with active = True or records with active = False.